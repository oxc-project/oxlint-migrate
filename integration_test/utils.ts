import { expect } from 'vitest';
import main from '../src/index.js';
import { OxlintConfig } from '../src/types.js';

export const getSnapshotResult = async (
  config: Parameters<typeof main>[0],
  oxlintConfig?: OxlintConfig
) => {
  const collector: string[] = [];
  const result = await main(config, oxlintConfig, {
    reporter: collector.push.bind(collector),
    upgrade: oxlintConfig !== undefined,
  });

  return {
    config: result,
    warnings: collector
      // filter out unsupported rules
      .filter((error) => !error.startsWith('unsupported rule: local/'))
      .filter((error) => !error.startsWith('unsupported rule: perfectionist/'))
      .filter((error) => !error.startsWith('unsupported rule: vue/'))
      .filter((error) => !error.startsWith('unsupported rule: regexp/'))
      .filter((error) => !error.startsWith('unsupported rule: toml/'))
      .filter((error) => !error.startsWith('unsupported rule: style/')),
  };
};

export const getSnapShotUpgradeResult = async (
  config: Parameters<typeof main>[0],
  oxlintConfig: OxlintConfig
) => {
  const result = await getSnapshotResult(config, oxlintConfig);
  const upgradedResult = structuredClone(result);
  const result2 = await getSnapshotResult(config, upgradedResult.config);

  expect(result2).toStrictEqual(result);

  return result2;
};
