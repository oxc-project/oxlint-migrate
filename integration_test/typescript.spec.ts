import { expect, test } from 'vitest';
// @ts-ignore
import typescript_test from './projects/typescript.eslint.config.mjs';
import { getSnapshotResult, getSnapShotUpgradeResult } from './utils.js';

test('typescript', async () => {
  const result = await getSnapshotResult(typescript_test);
  expect(result).toMatchSnapshot('typescript');
});

test('typescript upgrade', async () => {
  const result = await getSnapShotUpgradeResult(typescript_test, {
    categories: {
      correctness: 'error',
      perf: 'error',
    },
  });
  expect(result).toMatchSnapshot('typescript--upgrade');
});
