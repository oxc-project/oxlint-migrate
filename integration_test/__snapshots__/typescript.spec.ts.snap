// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`typescript > typescript 1`] = `
{
  "config": {
    "categories": {
      "correctness": "off",
    },
    "env": {
      "builtin": true,
      "commonjs": true,
      "node": true,
      "shared-node-browser": true,
    },
    "ignorePatterns": [
      "**/node_modules/**",
      "built/**",
      "tests/**",
      "lib/**",
      "src/lib/*.generated.d.ts",
      "scripts/**/*.js",
      "scripts/**/*.d.*",
      "internal/**",
      "coverage/**",
    ],
    "overrides": [
      {
        "files": [
          "**/*.mjs",
          "**/*.mts",
        ],
        "rules": {
          "no-restricted-globals": [
            "error",
            {
              "name": "__filename",
            },
            {
              "name": "__dirname",
            },
            {
              "name": "require",
            },
            {
              "name": "module",
            },
            {
              "name": "exports",
            },
          ],
        },
      },
      {
        "files": [
          "src/**",
        ],
        "rules": {
          "no-restricted-globals": [
            "error",
            {
              "name": "setTimeout",
            },
            {
              "name": "clearTimeout",
            },
            {
              "name": "setInterval",
            },
            {
              "name": "clearInterval",
            },
            {
              "name": "setImmediate",
            },
            {
              "name": "clearImmediate",
            },
            {
              "name": "performance",
            },
          ],
        },
      },
      {
        "files": [
          "src/harness/**",
          "src/testRunner/**",
        ],
        "rules": {
          "no-restricted-globals": "off",
        },
      },
      {
        "files": [
          "src/lib/*.d.ts",
        ],
        "rules": {
          "@typescript-eslint/no-unsafe-function-type": "off",
          "@typescript-eslint/no-unused-vars": "off",
          "@typescript-eslint/no-wrapper-object-types": "off",
          "@typescript-eslint/prefer-function-type": "off",
          "no-restricted-globals": "off",
          "no-shadow-restricted-names": "off",
          "no-var": "off",
        },
      },
      {
        "files": [
          "src/lib/es2019.array.d.ts",
        ],
        "rules": {
          "@typescript-eslint/array-type": "off",
        },
      },
    ],
    "plugins": [
      "typescript",
    ],
    "rules": {
      "@typescript-eslint/adjacent-overload-signatures": "error",
      "@typescript-eslint/array-type": "error",
      "@typescript-eslint/ban-ts-comment": "off",
      "@typescript-eslint/ban-tslint-comment": "error",
      "@typescript-eslint/consistent-generic-constructors": "off",
      "@typescript-eslint/consistent-indexed-object-style": "off",
      "@typescript-eslint/consistent-type-definitions": "error",
      "@typescript-eslint/no-array-constructor": "error",
      "@typescript-eslint/no-confusing-non-null-assertion": "error",
      "@typescript-eslint/no-duplicate-enum-values": "off",
      "@typescript-eslint/no-empty-function": "off",
      "@typescript-eslint/no-empty-interface": "off",
      "@typescript-eslint/no-empty-object-type": "off",
      "@typescript-eslint/no-explicit-any": "off",
      "@typescript-eslint/no-extra-non-null-assertion": "error",
      "@typescript-eslint/no-inferrable-types": "off",
      "@typescript-eslint/no-misused-new": "error",
      "@typescript-eslint/no-namespace": "off",
      "@typescript-eslint/no-non-null-asserted-optional-chain": "off",
      "@typescript-eslint/no-require-imports": "off",
      "@typescript-eslint/no-this-alias": "error",
      "@typescript-eslint/no-unnecessary-type-constraint": "error",
      "@typescript-eslint/no-unsafe-declaration-merging": "error",
      "@typescript-eslint/no-unsafe-function-type": "error",
      "@typescript-eslint/no-unused-expressions": [
        "error",
        {
          "allowTernary": true,
        },
      ],
      "@typescript-eslint/no-unused-vars": [
        "warn",
        {
          "argsIgnorePattern": "^(_+$|_[^_])",
          "varsIgnorePattern": "^(_+$|_[^_])",
        },
      ],
      "@typescript-eslint/no-var-requires": "off",
      "@typescript-eslint/no-wrapper-object-types": "error",
      "@typescript-eslint/prefer-as-const": "error",
      "@typescript-eslint/prefer-for-of": "error",
      "@typescript-eslint/prefer-function-type": "error",
      "@typescript-eslint/prefer-namespace-keyword": "error",
      "@typescript-eslint/triple-slash-reference": "error",
      "constructor-super": "error",
      "eqeqeq": "error",
      "for-direction": "error",
      "getter-return": "error",
      "no-array-constructor": "off",
      "no-async-promise-executor": "error",
      "no-caller": "error",
      "no-case-declarations": "off",
      "no-class-assign": "error",
      "no-compare-neg-zero": "error",
      "no-cond-assign": "off",
      "no-const-assign": "error",
      "no-constant-binary-expression": "error",
      "no-constant-condition": [
        "error",
        {
          "checkLoops": false,
        },
      ],
      "no-control-regex": "off",
      "no-debugger": "error",
      "no-delete-var": "error",
      "no-dupe-class-members": "error",
      "no-dupe-else-if": "error",
      "no-dupe-keys": "error",
      "no-duplicate-case": "error",
      "no-empty": "error",
      "no-empty-character-class": "off",
      "no-empty-function": "off",
      "no-empty-pattern": "error",
      "no-empty-static-block": "error",
      "no-eval": "error",
      "no-ex-assign": "error",
      "no-extra-boolean-cast": "off",
      "no-fallthrough": "error",
      "no-func-assign": "error",
      "no-global-assign": "error",
      "no-import-assign": "error",
      "no-inner-declarations": "off",
      "no-invalid-regexp": "off",
      "no-irregular-whitespace": "error",
      "no-loss-of-precision": "error",
      "no-new-func": "error",
      "no-new-native-nonconstructor": "error",
      "no-new-wrappers": "error",
      "no-nonoctal-decimal-escape": "error",
      "no-obj-calls": "error",
      "no-prototype-builtins": "error",
      "no-redeclare": "error",
      "no-regex-spaces": "error",
      "no-self-assign": "error",
      "no-setter-return": "error",
      "no-shadow-restricted-names": "error",
      "no-sparse-arrays": "error",
      "no-template-curly-in-string": "error",
      "no-this-before-super": "error",
      "no-throw-literal": "error",
      "no-undef": "error",
      "no-unexpected-multiline": "error",
      "no-unreachable": "error",
      "no-unsafe-finally": "error",
      "no-unsafe-negation": "error",
      "no-unsafe-optional-chaining": "error",
      "no-unused-expressions": "off",
      "no-unused-labels": "error",
      "no-unused-private-class-members": "error",
      "no-unused-vars": "off",
      "no-useless-catch": "error",
      "no-useless-escape": "error",
      "no-var": "error",
      "no-with": "error",
      "prefer-object-spread": "error",
      "require-yield": "error",
      "unicode-bom": [
        "error",
        "never",
      ],
      "use-isnan": "error",
      "valid-typeof": "error",
    },
  },
  "warnings": [
    "unsupported rule: no-dupe-args",
    "unsupported rule: no-misleading-character-class",
    "unsupported rule: no-octal",
    "unsupported rule: no-useless-backreference",
    "unsupported rule: prefer-const",
    "unsupported rule: prefer-const",
    "unsupported rule: @typescript-eslint/class-literal-property-style",
    "unsupported rule: @typescript-eslint/consistent-type-assertions",
    "unsupported rule: no-misleading-character-class",
    "unsupported rule: prefer-regex-literals",
    "unsupported rule: dot-notation",
    "unsupported rule: no-extra-bind",
    "unsupported rule: no-return-await",
    "unsupported rule: no-undef-init",
    "unsupported rule: object-shorthand",
    "unsupported rule: prefer-const",
    "unsupported rule: no-restricted-syntax",
    "unsupported rule: @typescript-eslint/naming-convention",
    "unsupported rule: @typescript-eslint/unified-signatures",
    "unsupported rule: @typescript-eslint/no-unnecessary-type-assertion",
    "unsupported plugin for rule: @typescript-eslint/no-unsafe-function-type",
    "unsupported plugin for rule: @typescript-eslint/no-wrapper-object-types",
    "unsupported plugin for rule: @typescript-eslint/no-unused-vars",
    "unsupported plugin for rule: @typescript-eslint/no-array-constructor",
    "unsupported plugin for rule: @typescript-eslint/no-duplicate-enum-values",
    "unsupported plugin for rule: @typescript-eslint/no-empty-object-type",
    "unsupported plugin for rule: @typescript-eslint/no-explicit-any",
    "unsupported plugin for rule: @typescript-eslint/no-extra-non-null-assertion",
    "unsupported plugin for rule: @typescript-eslint/no-misused-new",
    "unsupported plugin for rule: @typescript-eslint/no-namespace",
    "unsupported plugin for rule: @typescript-eslint/no-non-null-asserted-optional-chain",
    "unsupported plugin for rule: @typescript-eslint/no-require-imports",
    "unsupported plugin for rule: @typescript-eslint/no-this-alias",
    "unsupported plugin for rule: @typescript-eslint/no-unnecessary-type-constraint",
    "unsupported plugin for rule: @typescript-eslint/no-unsafe-declaration-merging",
    "unsupported plugin for rule: @typescript-eslint/no-unsafe-function-type",
    "unsupported plugin for rule: @typescript-eslint/no-unused-expressions",
    "unsupported plugin for rule: @typescript-eslint/no-unused-vars",
    "unsupported plugin for rule: @typescript-eslint/no-wrapper-object-types",
    "unsupported plugin for rule: @typescript-eslint/prefer-as-const",
    "unsupported plugin for rule: @typescript-eslint/prefer-namespace-keyword",
    "unsupported plugin for rule: @typescript-eslint/triple-slash-reference",
    "unsupported plugin for rule: @typescript-eslint/adjacent-overload-signatures",
    "unsupported plugin for rule: @typescript-eslint/array-type",
    "unsupported plugin for rule: @typescript-eslint/ban-tslint-comment",
    "unsupported plugin for rule: @typescript-eslint/consistent-generic-constructors",
    "unsupported plugin for rule: @typescript-eslint/consistent-indexed-object-style",
    "unsupported plugin for rule: @typescript-eslint/consistent-type-definitions",
    "unsupported plugin for rule: @typescript-eslint/no-confusing-non-null-assertion",
    "unsupported plugin for rule: @typescript-eslint/no-empty-function",
    "unsupported plugin for rule: @typescript-eslint/no-inferrable-types",
    "unsupported plugin for rule: @typescript-eslint/prefer-for-of",
    "unsupported plugin for rule: @typescript-eslint/prefer-function-type",
    "unsupported plugin for rule: @typescript-eslint/no-var-requires",
    "unsupported plugin for rule: @typescript-eslint/no-empty-interface",
  ],
}
`;
